tsconfig.json

```diff
@@ -7,6 +7,8 @@
     "strict": true,
     "noImplicitReturns": true,
     "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
```

Comment:
this should not be included: we intentionally avoid to allow synthetic default import

Reply:
This was added to support compile time typings (no emit) - only for `import pkg from 'shelljs'` which doesn't support. If I convert the integration tests setup to commonjs then I can likely use ts-node again.

Reply:
you still can import pkgs without this option. This is global option affects all srcs, so we'd like to avoid it.

Reply:
Makes sense, However I couldn't avoid it when importing commonjs into esm

Reply:
yes, agreed. that's what I encountered several time https://twitter.com/_ojkwon/status/1362187177975631873. Since we should not break any existing ergonomics, unlike greenfield we have lot of constraints to keep it. tsconfig is something falls into those category, especially we had some history with default import / export and explicitly banned out options around default import so we want to keep it while making changes.

Reply:
Changing this specific option doesn't have any effect on code emit - its only effect is the type checking that occurs within this project.

Reply:
Theoritically yes, but root option compiles down things to src shouldn't change due to other places. I personally against to emable this option in this repo, but if we can't avoid it'd better to separate config where it's being used.